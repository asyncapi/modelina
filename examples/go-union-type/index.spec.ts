const spy = jest.spyOn(global.console, 'log').mockImplementation(() => {
  return;
});
import { generate } from './index';

describe('Should be able to render union types', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    expect(spy.mock.calls.length).toEqual(6);
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[1]).toMatchSnapshot();
    expect(spy.mock.calls[2]).toMatchSnapshot();
    expect(spy.mock.calls[3]).toMatchSnapshot();
    expect(spy.mock.calls[4]).toMatchSnapshot();
    expect(spy.mock.calls[5]).toMatchSnapshot();
  });
});
