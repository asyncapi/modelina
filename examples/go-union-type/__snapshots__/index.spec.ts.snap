// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to render union types and should log expected output to console 1`] = `
Array [
  "// AdditionalProperty represents a AdditionalProperty model.
type AdditionalProperty struct {
  AdditionalPropertyOneOf_0
  AdditionalPropertyOneOf_1
  string
  float64
  ModelinaArrType []string
  ModelinaArrType []Union
  AdditionalPropertyOneOf_6
}",
]
`;

exports[`Should be able to render union types and should log expected output to console 2`] = `
Array [
  "// Union represents a Union model.
type Union struct {
  string
  float64
  ModelinaAnyType interface{}
}",
]
`;

exports[`Should be able to render union types and should log expected output to console 3`] = `
Array [
  "// AdditionalPropertyOneOf_6 represents a AdditionalPropertyOneOf_6 model.
type AdditionalPropertyOneOf_6 struct {
  string
  float64
  bool
}",
]
`;

exports[`Should be able to render union types and should log expected output to console 4`] = `
Array [
  "",
]
`;

exports[`Should be able to render union types and should log expected output to console 5`] = `
Array [
  "// AdditionalPropertyOneOf_0 represents a AdditionalPropertyOneOf_0 model.
type AdditionalPropertyOneOf_0 struct {
  Ref string
  AdditionalProperties map[string]interface{}
}",
]
`;

exports[`Should be able to render union types and should log expected output to console 6`] = `
Array [
  "// AdditionalPropertyOneOf_1 represents a AdditionalPropertyOneOf_1 model.
type AdditionalPropertyOneOf_1 struct {
  Id string
  AdditionalProperties map[string]interface{}
}",
]
`;
